### PR TITLE
model building steps with micrograd - karpathy

### PR DESCRIPTION
**Pull request by mratanusarkar**
Building a dummy mlp model demonstrating a neural network model building steps:

- dataset
- loss function
- optimization & gradient descent
- training steps

Also, updates micrograd module with a collection mechanism for model parameters. (#6)

### CODE REVIEW

**Review by mratanusarkar — COMMENTED** [review]
finally, an end to the first video in the series!

TODO:
- need to add zero grad.
- add more features to the micrograd module.
- also, typedef, docs, and readme to be added.
